tests/Unit/API/Trace/TraceStateTest.php

```diff
@@ -210,4 +213,58 @@ public function test_validate_value(): void
         $this->assertSame(257, strlen($invalidValue));
         $this->assertNull($tracestate->get($validKey));
     }
+
+    #[DataProvider('toStringProvider')]
+    public function test_to_string(array $entries, ?int $limit, string $expected): void
+    {
+        $traceState = new TraceState();
+        foreach (array_reverse($entries) as $key => $value) {
+            $traceState = $traceState->with($key, $value);
+        }
+
+        $this->assertSame($expected, $traceState->toString(limit: $limit));
+    }
+
+    public static function toStringProvider(): array
+    {
+        return [
```

Comment:
Perhaps add test for value ending with space here too (which would be rejected), as that doesn't seem to be covered by other tests.

Reply:
Will readd https://github.com/open-telemetry/opentelemetry-php/pull/1267/commits/bace440ea13a17c0bd2f08da99f0e728a474f7d7#diff-48f55048bb7253924533e9078b9eda9994babffa4ef746e8a16a65d65a0dc319L162-L181, which includes a test for value with trailing space.